src/actions/WalletActions.tsx

```diff
@@ -147,101 +147,98 @@ export function updateMostRecentWalletsSelected(walletId: string, tokenId: EdgeT
-  if (assetOptions.length === 1 && assetOptions[0].paymentWalletId != null) {
-    const { paymentWalletId, tokenId } = assetOptions[0]
-    const activationQuote = await account.activateWallet({
-      activateWalletId: wallet.id,
-      activateTokenIds: tokenIds,
-      paymentInfo: {
-        walletId: paymentWalletId,
-        tokenId
-      }
-    })
-    const tokensText = tokenIds.map(tokenId => {
-      const { currencyCode, displayName } = getToken(wallet, tokenId) ?? {}
-      return `${displayName} (${currencyCode})`
-    })
-    const tileTitle = tokenIds.length > 1 ? lstrings.activate_wallet_tokens_scene_tile_title : lstrings.activate_wallet_token_scene_tile_title
-    const tileBody = tokensText.join(', ')
-
-    const { networkFee } = activationQuote
-    const { nativeAmount: nativeFee, currencyPluginId, tokenId: feeTokenId } = networkFee
-    if (currencyPluginId !== pluginId) throw new Error('Internal Error: Fee asset mismatch.')
-
-    const paymentCurrencyCode = getCurrencyCode(wallet, feeTokenId)
-
-    const exchangeNetworkFee = await wallet.nativeToDenomination(nativeFee, paymentCurrencyCode)
-    const feeDenom = selectDisplayDenomByCurrencyCode(state, wallet.currencyConfig, paymentCurrencyCode)
-    const displayFee = div(nativeFee, feeDenom.multiplier, log10(feeDenom.multiplier))
-    let fiatFee = convertCurrencyFromExchangeRates(state.exchangeRates, paymentCurrencyCode, defaultIsoFiat, exchangeNetworkFee)
-    if (lt(fiatFee, '0.001')) fiatFee = '<0.001'
-    else fiatFee = round(fiatFee, -3)
-    const feeString = `${displayFee} ${feeDenom.name} (${fiatFee} ${defaultFiat})`
-    let bodyText = lstrings.activate_wallet_token_scene_body
-
-    const { tokenActivationAdditionalReserveText } = SPECIAL_CURRENCY_INFO[pluginId] ?? {}
-    if (tokenActivationAdditionalReserveText != null) {
-      bodyText += '\n\n' + tokenActivationAdditionalReserveText
-    }
-
-    navigation.navigate('confirmScene', {
-      titleText: lstrings.activate_wallet_token_scene_title,
-      bodyText,
-      infoTiles: [
-        { label: tileTitle, value: tileBody },
-        { label: lstrings.mining_fee, value: feeString }
-      ],
-      onConfirm: (resetSlider: () => void) => {
-        if (lt(wallet.balanceMap.get(feeTokenId) ?? '0', nativeFee)) {
-          const msg = tokenIds.length > 1 ? lstrings.activate_wallet_tokens_insufficient_funds_s : lstrings.activate_wallet_token_insufficient_funds_s
-          Airship.show<'ok' | undefined>(bridge => (
-            <ButtonsModal
-              bridge={bridge}
-              title={lstrings.create_wallet_account_unfinished_activation_title}
-              message={sprintf(msg, feeString)}
-              buttons={{ ok: { label: lstrings.string_ok } }}
-            />
-          )).catch(err => showError(err))
-          navigation.pop()
-          return
+function activateWalletTokens(navigation: NavigationBase, wallet: EdgeCurrencyWallet, tokenIds?: string[]): ThunkAction<Promise<void>> {
```

Comment:
It seems like the `tokenIds` argument should be mandatory, and should be `EdgeTokenId[]`. The function is only called from two places, so this change should be safe. Within the function, we pass `tokenIds` straight to the core, so it's up to the plugins to decide if they support mainnet activation or not.

Further down, when we do `const tokensText`, we could change the following line to `const { currencyCode, displayName } = tokenId != null ? wallet.currencyConfig.allTokens[tokenId] : wallet.currencyInfo`, and even our string formatting will work for mainnet coins.